clab/general.go

```diff
@@ -0,0 +1,130 @@
+package clab
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/srl-labs/containerlab/types"
+	"github.com/srl-labs/containerlab/utils"
+)
+
+const (
+	clabHostEntryPrefix  = "###### CLAB-%s-START ######"
+	clabHostEntryPostfix = "###### CLAB-%s-END ######"
+)
+
+func AppendHostsFileEntries(containers []types.GenericContainer, labname string) error {
+	filename := "/etc/hosts"
+	if labname == "" {
+		return fmt.Errorf("missing lab name")
+	}
+	if !utils.FileExists(filename) {
+		err := utils.CreateFile(filename, "127.0.0.1\tlocalhost")
+		if err != nil {
+			return err
+		}
+	}
+	// lets make sure we do not have remaining of a non destroyed run in the hosts file
+	err := DeleteEntriesFromHostsFile(labname)
+	if err != nil {
+		return err
+	}
+	data := generateHostsEntries(containers, labname)
+	if len(data) == 0 {
+		return nil
+	}
+	var f *os.File
+
+	f, err = os.OpenFile("/etc/hosts", os.O_APPEND|os.O_WRONLY, os.ModeAppend)
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+	_, err = f.Write(data)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// hostEntries builds an /etc/hosts compliant text blob (as []byte]) for containers ipv4/6 address<->name pairs
+func generateHostsEntries(containers []types.GenericContainer, labname string) []byte {
+
+	entries := bytes.Buffer{}
+	v6entries := bytes.Buffer{}
+
+	fmt.Fprintf(&entries, clabHostEntryPrefix, labname)
+	entries.WriteByte('\n')
+
+	for _, cont := range containers {
+		if len(cont.Names) == 0 {
+			continue
+		}
+		if cont.NetworkSettings.Set {
+			if cont.NetworkSettings.IPv4addr != "" {
+				fmt.Fprintf(&entries, "%s\t%s\n", cont.NetworkSettings.IPv4addr, strings.TrimLeft(cont.Names[0], "/"))
+			}
+			if cont.NetworkSettings.IPv6addr != "" {
+				fmt.Fprintf(&v6entries, "%s\t%s\n", cont.NetworkSettings.IPv6addr, strings.TrimLeft(cont.Names[0], "/"))
+			}
+		}
+	}
```

Comment:
maybe I am missing something here, but why don't use the same `entries` and write IPv6addr to it in the same loop?

Reply:
If I'm not mistaking also you liked the idea to have IP versions separated in the block.
It helps a lot with readability should someone take a look at the hosts file for reference.
In the end this was my personal liking to do it that way.

Reply:
ah, I see. you group ipv4 and ipv6 separately inside the same lab hosts block

I didn't get this intention from looking at the code.
Do you happen to have the example of the hosts content with such grouping (I have no lab setup with me at the moment)

Reply:
Sorry wasn't you it was @kellerza -> https://github.com/srl-labs/containerlab/pull/520#discussion_r672846069

Reply:
Indeed, I like separating them. 

Maybe it's OCD, but the length difference between IPv4&6 messes up the alignment if you don't group them

```
172.20.20.2     clab-tag2-sr2
2001:172:20:20::2       clab-tag2-sr2
172.20.20.3     clab-tag2-sr1
2001:172:20:20::3       clab-tag2-sr1
```